src/deepsparse/transformers/pipelines/text_generation/nl_engine_operator.py

```diff
@@ -208,14 +209,15 @@ def run(self, inp: NLEngineInputs, **kwargs) -> NLEngineOutputs:
             # (batch_size=0) to the engine. Therefore,
             # we skip the validation
 
-            # Internal kv_cache works for batch_size of 1 atm
-            out = self.engine._eng_net.execute_list_out(
-                inputs[0], kv_cache[0].engine_internal_cache
-            )
+            internal_kv_cache = [x.engine_internal_cache for x in kv_cache]
+            if inp.engine:
```

Comment:
`inp.engine`? As in the engine is an attribute of the input and not the operator itself? When does this happen?

Reply:
When we introduced continuous_batching, this was added in so that we could use the engine which has the model with the correct batch size